### PR TITLE
📖Add instructions for providers to upgrade to CRD v1

### LINKED ISSUE #2546: Document CRDv1 requirement for providers
All Cluster API providers must upgrade to CRDv1 and controller-runtime v0.5.x, we need to add documentation to the [v1alpha2 to v1alpha3 document](https://master.cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html) that provides steps to perform the upgrade.

/kind documentation
/priority critical-urgent
/milestone v0.3.0

### PR DESCRIPTION
**Pull request by nader-ziada**
**What this PR does / why we need it**:
- Add instructions in docs for providers to upgrade to CRD v1

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2546

### CODE REVIEW

File: docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md [hunk 1]
```diff
@@ -371,6 +371,83 @@ commonLabels:
   cluster.x-k8s.io/v1beta1: v1alphaX,v1beta1
 ```
 
+# Upgrade to CRD v1
+
+- Providers should upgrade their CRDs to v1
+- Minimum Kubernrtes version should be `v1.16.x`
```
Comment (vincepri) [inline docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md]:
```suggestion
- Minimum Kubernetes version supporting CRDv1 is `v1.16`
```

File: docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md [hunk 2]
```diff
@@ -371,6 +371,83 @@ commonLabels:
   cluster.x-k8s.io/v1beta1: v1alphaX,v1beta1
 ```
 
+# Upgrade to CRD v1
```
Comment (vincepri) [inline docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md]: Should we put this at the very top? Seems might be the first thing to tackle probably

Comment (nader-ziada) [reply docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md]: the item right before it `update config folder` is a prerequisite, so I put it right after that one

File: docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md [hunk 3]
```diff
@@ -371,6 +371,83 @@ commonLabels:
...
+    output:crd:dir=$(CRD_ROOT) \
+    output:webhook:dir=$(WEBHOOK_ROOT) \
+    webhook
+  $(CONTROLLER_GEN) \
+    paths=./controllers/... \
+    output:rbac:dir=$(RBAC_ROOT) \
+    rbac:roleName=manager-role
+```
+
+- For all the CRDs in the `config/crd/bases` change the version of `CustomResourceDefinetioin` to `v1`
```
Comment (ncdc) [inline docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md]:
```suggestion
- For all the CRDs in the `config/crd/bases` change the version of `CustomResourceDefinition` to `v1`
```

**Pull request merged**